src/main/java/edu/ucsb/cs156/dining/controllers/MenuItemReviewController.java

```diff
@@ -0,0 +1,99 @@
+import edu.ucsb.cs156.dining.errors.EntityNotFoundException;
+import edu.ucsb.cs156.dining.repositories.MenuItemReviewRepository;
+import edu.ucsb.cs156.dining.repositories.MenuItemRepository;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import lombok.extern.slf4j.Slf4j;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.server.ResponseStatusException;
+import org.springframework.http.HttpStatus;
+
+import jakarta.validation.Valid;
+import java.time.LocalDateTime;
+
+
+/**
+ * This is a REST controller for MenuItemReview
+ */
+
+@Tag(name = "MenuItemReview")
+@RequestMapping("/api/menuitemreviews")
+@RestController
+@Slf4j
+public class MenuItemReviewController extends ApiController {
+
+    @Autowired
+    MenuItemReviewRepository menuItemReviewRepository;
+
+    @Autowired
+    MenuItemRepository menuItemRepository;
+    
+    /**
+     * Create a new menu item review -> all users
+     */
+
+    @Operation(summary= "Create a new menu item review")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @PostMapping("/post")
+    public MenuItemReview postMenuItemReview(
+            @Parameter(name="studentUserId") @RequestParam long studentUserId,
+            @Parameter(name="itemId") @RequestParam long itemId,
+            @Parameter(name="itemServedDate", description="date (in iso format, e.g. YYYY-mm-ddTHH:MM:SS") @RequestParam("itemServedDate") @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime itemServedDate,
+            @Parameter(name="status") @RequestParam String status,
+            @Parameter(name="moderatorUserId") @RequestParam long moderatorUserId,
+            @Parameter(name="moderatorComments") @RequestParam String moderatorComments,
+            @Parameter(name="createdDate", description="date (in iso format, e.g. YYYY-mm-ddTHH:MM:SS") @RequestParam("createdDate") @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime createdDate,
+            @Parameter(name="lastEditedDate", description="date (in iso format, e.g. YYYY-mm-ddTHH:MM:SS") @RequestParam("lastEditedDate") @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime lastEditedDate)
```

Comment:
```suggestion
           
```

Last edited date should be set either automatically using the techniques here:

https://ucsb-cs156.github.io/topics/spring_boot/spring_boot_database.html#timestamps-on-database-rows

Or it should be set inside the controller in the POST and PUT methods, plus any method that the moderator uses to change the status.

Allowing the *user* to input this means they can choose any date/time they want, in the past, present or future.  What would be the purpose of the field if the user can make it say anything they want?

It's so important to keep the use case and the user story in mind when writing code for real applications!